### ISSUE TITLE
Failure to expand tilde to resolve home

### ISSUE THREAD

**Issue by russauld** [sethvargo/chef-api#54]
The following error occurs when I run my script from within another tool. I'm not exactly sure how it calls the script, but it consistently fails like this:
```
Error: headnode: remove_chef_node.rb: return code=1. Error message=/opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/defaults.rb:27:in `expand_path': couldn't find login name -- expanding `~' (ArgumentError)
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/defaults.rb:27:in `config'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/defaults.rb:48:in `endpoint'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/defaults.rb:19:in `block in options'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/defaults.rb:19:in `map'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/defaults.rb:19:in `options'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/configurable.rb:59:in `block in reset!'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/configurable.rb:58:in `each'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api/configurable.rb:58:in `reset!'
        from /opt/ruby/2.4.0/lib/ruby/gems/2.4.0/gems/chef-api-0.7.0/lib/chef-api.rb:98:in `<top (required)>'
        from /opt/ruby/2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:133:in `require'
        from /opt/ruby/2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:133:in `rescue in require'
        from /opt/ruby/2.4.0/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:40:in `require'
        from /install/prescripts/remove_chef_node.rb:10:in `<main>'
```
When I run the script from the command line this error does not happen.
I'll submit a PR to fix this - we just need to replace `'~/.chef-api'` with `"#{ENV['HOME']}/.chef-api"`

**Comment by russauld** [comment]
This appears to be related to issue #47 - will test that fix and report back.
FWIW, I did a 'gem install chef-api' and that version didn't pick up the change added by the PR that fixes issue #47

**Comment by tas50** [comment]
This should be fixed in 0.7.1 which I just pushed up
